universal-application-tool-0.0.1/app/services/applicant/question/FileUploadQuestion.java

```diff
@@ -18,6 +18,7 @@
 
   private final ApplicantQuestion applicantQuestion;
   private Optional<String> fileKeyValue;
+  private Optional<String> originalFileNameValue;
```

Comment:
Given the usage these should be `Optional<Optional<String>>`.  and initialized in the constructor.  Add a comment to indicate that this is serving double duty as a singleton load of the value.

I don't think we have a standard on using nulls or not, but I don't think we _need_ to use nulls here.

Reply:
I've updated, do you mind taking a look if I implemented it the way you had in mind? I got a little lost in all the optionals haha